source/slang/slang-ir-legalize-varying-params.cpp

```diff
@@ -1512,10 +1514,21 @@ struct CPUEntryPointVaryingParamLegalizeContext : EntryPointVaryingParamLegalize
 
         dispatchThreadID =
             emitCalcDispatchThreadID(builder, uint3Type, groupID, groupThreadID, groupExtents);
+        UInt idx = 0;
+        dispatchThreadID1D = builder.emitSwizzle(uintType, dispatchThreadID, 1, &idx);
```

Comment:
This isn't general enough to handle the cases like `int2 tid : SV_DispatchThreadID` or 
`vector<uint8_t, 2> tid : SV_DispatchThreadID`.

We will need the same type adaptor mechanism implemented for other targets here, and same applies to group id and other builtins as well.